autorest/m4reformatter/__init__.py

```diff
@@ -489,11 +492,20 @@ def update_overloads(
             overload = self.update_overload(
                 group_name, yaml_data, body_type, content_types=content_types
             )
-            for parameter in overload["parameters"]:
-                if parameter["restApiName"] == "Content-Type":
-                    parameter["clientDefaultValue"] = overload["bodyParameter"][
-                        "defaultContentType"
-                    ]
+            for param in overload["parameters"]:
```

Comment:
you shouldn't need all this code, and to add `checkContentTypes`. For the `RequestBody`, we already include all of the content types associated with it, so you shouldn't need extra logic here adding that to `checkContentTypes`. All you should need to do is make sure the content type param is still required in this scenario